administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
+            $src_frames[$key]['image'] = imagerotate($src_frames[$key]['image'], $angle, 0);
           }
-          else
-          {
-            $src_img = imagecreatefromgif($src_file);
-          }
-        }
-        if(!$src_img)
-        {
-          return false;
         }
 
-        if($angle > 0)
+        foreach ($src_frames as $key => $frame)
         {
-          $src_img = imagerotate($src_img, $angle, 0);
+          if (!is_null($offsetx) && !is_null($offsety))
+          // resizing with GD1
+          {
+            imagecopyresized( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, $offsetx, $offsety,
+                              $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
+          }
+          else
+          {
+            imagecopyresized( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, 0, 0, $destWidth,
+                              (int)$destHeight, $srcWidth, $srcHeight);
+          }
+          // write resized image to file
+          $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_frames,$dest_qual,$dest_imgtype);
         }
 
-        $dst_img = imagecreate($destWidth, $destHeight);
-
-        if (!is_null($offsetx) && !is_null($offsety))
+        if ($metadata)
+        // copy metadata if needed
         {
-          imagecopyresized( $dst_img, $src_img, 0, 0, $offsetx, $offsety,
-                            $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
-        }
-        else
-        {
-          imagecopyresized( $dst_img, $src_img, 0, 0, 0, 0, $destWidth,
-                            (int)$destHeight, $srcWidth, $srcHeight);
+          $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+          if (!$meta_success)
+          {
+            $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+            return false;
+          }      
         }
-        if(!@imagejpeg($dst_img, $dest_file, $dest_qual))
+        if(!$success)
         {
           // Workaround for servers with wwwrun problem
           $dir = dirname($dest_file);
           JoomFile::chmod($dir, '0777', true);
-          imagejpeg($dst_img, $dest_file, $dest_qual);
+          $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_img,$dest_qual,$dest_imgtype);
+          if ($metadata)
+          {
+            $meta_success = JoomFile::copyImageMetadata($src_file, $dest_file, $src_imagetype, $dest_imgtype);
+            if (!$meta_success)
+            {
+              $debugoutput.=JText::_('COM_JOOMGALLERY_UPLOAD_GD_ERROR_COPY_METADATA');
+              return false;
+            }      
+          }
           JoomFile::chmod($dir, '0755', true);
         }
-        imagedestroy($src_img);
-        imagedestroy($dst_img);
+        foreach ($src_frames as $key => $frame)
+        {
+          imagedestroy($src_frames[$key]['image']);
+          imagedestroy($dst_frames[$key]);
+        }
         break;
       case 'gd2':
```

Comment:
everything is the same as GD1. Differences:
- imagecreate becomes imagecreatetruecolor
- imagecopyresized becomes imagecopyresampled